pages/desktop/user.py

```diff
@@ -119,8 +124,28 @@ def click_update_account(self):
         self.selenium.find_element(*self._update_account_locator).click()
         WebDriverWait(self.selenium, self.timeout).until(lambda s: self.update_message == "Profile Updated")
 
-    def change_hide_email_state(self):
-        self.selenium.find_element(*self._hide_email_checkbox).click()
+    def hide_email_address(self):
+        email_checkbox = self.selenium.find_element(*self._email_checkbox)
+        WebDriverWait(self.selenium, self.timeout).until(expected_conditions.visibility_of(email_checkbox));
+
+        if email_checkbox.is_selected():
+            err_msg = 'Email is already hidden'
+            return err_msg
```

Comment:
You cannot return an error message here as nothing will be done with it. If you look at the calling code you wrote (https://github.com/mozilla/Addon-Tests/pull/765/files#diff-2e5bbbef9e51c633c062636e9f31733fR94), it just calls the method but doesn't do anything with the return value. So even if this code was encountered, and you returned `err_msg`, nothing would happen - the test would just continue on its merry way.

What you need to do is throw an exception here, which will stop the test immediately, as that's really what we want to happen if our expectations are not met. You could use:

`raise Exception('Email is already hidden')`